src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java

```diff
@@ -813,7 +842,19 @@ private long getOffset() {
 
       // This api should throw exception if backoff failed.
       // fileNamesCopy after this call is emptied (clears the input list)
-      ingestionService.ingestFiles(fileNamesCopy);
+      if (ingestionDeliveryGuarantee
+          == SnowflakeSinkConnectorConfig.IngestionDeliveryGuarantee.EXACTLY_ONCE) {
+        ingestionService.ingestFilesWithClientInfo(fileNamesCopy, this.clientSequencer.get());
+        String offsetToken = ingestionService.getClientStatus().getOffsetToken();
+        // Update server side offset
+        if (offsetToken == null) {
+          this.offsetPersistedInSnowflake.set(-1);
```

Comment:
Is this even possible? Should we throw an exception if offsetToken we receive from SF is null?

Reply:
when we create new pipe, the offset token is null when we call get pipe status

Reply:
we can not put null offset token in insert call though

Reply:
> when we create new pipe, the offset token is null when we call get pipe status

This will not happen in this call method right? because we are calling `ingestFilesWithClientInfo`